### DISCUSSION TITLE
Installation with scripts fails on debian 10 server

### DISCUSSION THREAD

**Discussion by ourocsoc** [searx/searx#2017] [Installation issue: Official wiki]
I follow the installation instructions in the wiki (except I don't have a user user, so I do it with the root user), and I get the following output:

```
======================
searx@207.246.127.246 
======================

Install searx@207.246.127.246  (service)
========================================

installation of packages
------------------------

package(s)::

   virtualenv python3-dev python3-babel python3-venv uwsgi
   uwsgi-plugin-python3 git build-essential libxslt-dev zlib1g-dev
   libffi-dev libssl-dev shellcheck

Should packages be installed? [YES/no] 
Yes
Hit:1 http://security.debian.org/debian-security buster/updates InRelease
Hit:2 http://deb.debian.org/debian buster InRelease
Hit:3 http://deb.debian.org/debian buster-updates InRelease
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'libxslt1-dev' instead of 'libxslt-dev'
build-essential is already the newest version (12.6).
git is already the newest version (1:2.20.1-2+deb10u3).
libffi-dev is already the newest version (3.2.1-9).
libxslt1-dev is already the newest version (1.1.32-2.2~deb10u1).
libssl-dev is already the newest version (1.1.1d-0+deb10u3).
python3-babel is already the newest version (2.6.0+dfsg.1-1).
virtualenv is already the newest version (15.1.0+ds-2).
python3-dev is already the newest version (3.7.3-1).
python3-venv is already the newest version (3.7.3-1).
shellcheck is already the newest version (0.5.0-3).
uwsgi is already the newest version (2.0.18-1).
uwsgi-plugin-python3 is already the newest version (2.0.18-1).
zlib1g-dev is already the newest version (1:1.2.11.dfsg-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
** press any [KEY] to continue **

user searx
----------

useradd --shell /bin/bash --system  --home-dir "/usr/local/searx"  --comment 'Privacy-respecting metasearch engine' searx
useradd: user 'searx' already exists
mkdir "/usr/local/searx"
mkdir: cannot create directory '/usr/local/searx': File exists
chown -R "searx:searx" "/usr/local/searx"
groups searx
-->|searx : searx
** press any [KEY] to continue **

Clone searx sources
-------------------

ls: cannot access '/root/searx': Permission denied
ERROR: ./utils/searx.sh: line 366: user 'searx' missed read permission: /root/searx
```

This appears be due to REPO_NAME being incorrectly set, but I cannot find where it is set.

**Comment by return42** [comment]
ERROR: ./utils/searx.sh: line 366: user 'searx' missed read permission: /root/searx 

[The scripts are creating a searx, a filtron and a morty user. These user do not have read access to /root and /root/searx folder](https://github.com/asciimoo/searx/pull/1803#issuecomment-618313334).
 
related: https://github.com/asciimoo/searx/issues/2013#issuecomment-647132313

**Comment by Monklite** [comment]
Same issue for me. I think it also failed on Ubuntu.

**Comment by return42** [comment]
> Same issue for me.

You installed as root? From folder /root/foo? .. what do you suggest to handle your issue?

> I think it also failed on Ubuntu.

The scripts are tested in the [LXC searx-suite](https://asciimoo.github.io/searx/utils/lxc.sh.html#searx-suite) on ubuntu (16.04, 18.04, 19.04), fedora31 and archlinux.

What exactly have you tested?

**Comment by Monklite** [comment]
> > Same issue for me.
> 
> You installed as root? From folder /root/foo? .. what do you suggest to handle your issue?
> 
> > I think it also failed on Ubuntu.
> 
> The scripts are tested in the [LXC searx-suite](https://asciimoo.github.io/searx/utils/lxc.sh.html#searx-suite) on ubuntu (16.04, 18.04, 19.04), fedora31 and archlinux.
> 
> What exactly have you tested?

I'm running
<code>sudo -H ./utils/searx.sh install all</code>  
per the installation instructions for the install scripts.
I'm running it as a root user from /root/searx (the cloned git repo)

**Comment by return42** [comment]
Tanks for the info ..

> I'm running it as a root user from /root/searx (the cloned git repo)

Read above: [The scripts are creating a searx, a filtron and a morty user. These user do not have read access to /root and /root/searx folder](https://github.com/asciimoo/searx/pull/1803#issuecomment-618313334).

Please login as normal user .. clone to folder which is readable for all users e.g. clone into the users ~/Downloads folder and than run the sudo commands from the docs .. the /root folder is only readable for root!

Why do you call sudo when you are root, normally this is pointless.

**Comment by Monklite** [comment]
> Tanks for the info ..
> 
> > I'm running it as a root user from /root/searx (the cloned git repo)
> 
> Read above: [The scripts are creating a searx, a filtron and a morty user. These user do not have read access to /root and /root/searx folder](https://github.com/asciimoo/searx/pull/1803#issuecomment-618313334).
> 
> Please login as normal user .. clone to folder which is readable for all users e.g. clone into the users ~/Downloads folder and than run the sudo commands from the docs .. the /root folder is only readable for root!
> 
> Why do you call sudo when you are root, normally this is pointless.

It worked, thanks.

**Comment by return42** [comment]
Thanks for your feedback!

@ourocsoc and @legofarme - What do you think, should I add a note to the installation instructions about the /root problem?

**Comment by Monklite** [comment]
> Thanks for your feedback!
> 
> @ourocsoc and @legofarme - What do you think, should I add a note to the installation instructions about the /root problem?

It wouldn't hurt, just a bolded note would do. I made a user called homeuser, created a Downloads folder, and put the git clone there.

**Comment by return42** [comment]
> It wouldn't hurt,

OK, I will left this issue open until I have made a node / thanks!

**Comment by ah1102** [comment]
![Снимок экрана 2020-07-21 в 17 58 26](https://user-images.githubusercontent.com/58341369/88073862-596fb680-cb7f-11ea-978b-ed00a3f0fce6.png)


I managed to solve the problem like this: `chown -R "searx: searx" "/root/"`

**Comment by return42** [comment]
> I managed to solve the problem like this: chown -R "searx: searx" "/root/"

**Do not change the owner of /root  !!!**

Create a folder anywhere (e.g. under /tmp) and give read access for all.

**Comment by ah1102** [comment]
Yes, after this command, I lost authorization with the server via ssh.

**Comment by hereticerik** [comment]
Also having this problem. This thread is 2 years old too... You'd think there'd be a simple solution in clear understandable English after 2 years.
